game.exs

```diff
@@ -1,7 +1,58 @@
 defmodule GuessingGame do
 
-  def hi() do
-    IO.puts("hello world")
+  def compare(n, t) do
+    # Best Practice ?
+
+    # a)
+    # case n do
+    #   m when m < t -> :low
+    #   m when m == t -> :equal
+    #   m when m > t -> :high
+    # end
+
+    # b)
+    cond do
+      n < t -> :low
+      n == t -> :equal
+      n > t -> :high
+    end
+  end
+
+  def print_state(state) do
+    case state do
+      :low -> IO.puts("Too low")
+      :equal -> IO.puts("That's the right number, you win!")
+      :high -> IO.puts("Too high")
+    end
+  end
+
+  def print_turns(turns) do
```

Comment:
Take a look at elixir's pattern matching. You could define `print_turns` twice — one that matches when turns is 0, and one that matches everything else. Their implementations will differ accordingly.